src/python/pants/option/option_types.py

```diff
@@ -660,19 +664,52 @@ def _convert_(self, val: Any) -> dict[str, _ValueT]:
 # -----------------------------------------------------------------------------------------------
 # "Specialized" Concrete Option Classes
 # -----------------------------------------------------------------------------------------------
+def _get_tool_name(subsystem_cls: type["Subsystem"]):
+    return getattr(subsystem_cls, "name", subsystem_cls.options_scope)
 
 
-class ArgsListOption(ShellStrListOption):
-    """A homogenous list of shell str options, to be used as arguments passed to some other tool.
+class SkipOption(BoolOption[bool]):
+    """A --skip option (for an invocable tool)."""
+
+    def __new__(cls, goal: str, *other_goals: str):
+        goals = (goal,) + other_goals
+        invocation_str = " and ".join([f"`{bin_name()} {goal}`" for goal in goals])
+        return super().__new__(
+            cls,  # type: ignore[arg-type]
+            "--skip",
+            default=False,  # type: ignore[arg-type]
+            help=(
+                lambda subsystem_cls: (
+                    f"Don't use {_get_tool_name(subsystem_cls)} when running {invocation_str}."
+                )
+            ),
+        )
 
-    Clients can call `passthrough()` to set the "passthrough" flag. See `passthrough` for more info.
-    """
 
-    def __new__(cls, help: _HelpT, *, passthrough: bool | None = None):
+class ArgsListOption(ShellStrListOption):
+    """An option for arguments passed to some other tool."""
+
+    def __new__(
+        cls,
+        *,
+        example: str,
+        extra_help: str = "",
+        tool_name: str | None = None,
+        # This should be set when callers can alternatively use "--" followed by the arguments,
+        # instead of having to provide "--[scope]-args='--arg1 --arg2'".
+        passthrough: bool | None = None,
```

Comment:
Why make this trinary? Ack it already was. Seems like `passthrough: bool = False` will work.

Reply:
I would, but the `None` is here for two reasons:
- The underlying options machinery doesn't like it when you pass certain `kwargs` in certain scenarios. This way models the current behavior of only adding the kwarg if it's set.
- I don't want to copy any assumptions from the options system until we start that process. Assuming `passthrough` is `False` leaks a little of that copying. When we're ready, it'll definitely be changed.